packages/flutter_driver/lib/src/driver.dart

```diff
@@ -205,6 +208,61 @@ class FlutterDriver {
     return result.text;
   }
 
+  /// Starts recording performance traces.
+  Future<Null> startTracing() async {
+    try {
+      await _appIsolate.invokeExtension(_kStartTracingMethod, const {});
+      return null;
+    } catch(error, stackTrace) {
+      throw new DriverError(
+        'Failed to start tracing due to remote error',
+        error,
+        stackTrace
+      );
+    }
+  }
+
+  /// Stops recording performance traces and downloads the trace profile.
+  Future<Map> stopTracingAndDownloadProfile() async {
+    Map<String, dynamic> stopResult =
+        await _appIsolate.invokeExtension(_kStopTracingMethod, const {});
+    String traceFilePath = stopResult['trace_file_path'];
+
+    // Tracing data isn't available immediately as some of it is queued up in
+    // the event loop.
+    Stopwatch sw = new Stopwatch()..start();
+    while(sw.elapsed < const Duration(seconds: 30)) {
+      Map<String, dynamic> downloadResult =
+          await _appIsolate.invokeExtension(_kDownloadTraceDataMethod, {
+            'trace_file_path': traceFilePath,
+          });
+
+      if (downloadResult['success'] == false) {
+        throw new DriverError('Failed to download trace file: $traceFilePath');
+      } else if (downloadResult['status'] != 'not ready') {
+        return downloadResult;
```

Comment:
this is the kind of case where I think we should be more aggressively parsing the JSON rather than handing unparsed JSON objects around (unparsed as in, it's been converted from UTF-8 bytes to Unicode JSON string to nested Maps, but still doesn't have intrinsic semantics). Doesn't have to be in this PR but I think that it would simplify the API if we weren't passing `Map<String, dynamic>` objects around so much.